### ISSUE TITLE
how to set MRL in llamacpp?

### ISSUE THREAD

**Issue by cheepon** [QwenLM/Qwen3-Embedding#5]
I found nothing about MRL support in llamacpp

Model: Qwen/Qwen3-Embedding-0.6B-GGUF

**Comment by afalf** [comment]
Sorry, llama.cpp does not currently support the MRL (Multi-Round Length) feature.  You can extract the unnormalized embedding from llama.cpp's output and manually adjust (truncate or project) the embedding dimensions as needed.

**Comment by afalf** [comment]
> > Sorry, llama.cpp does not currently support the MRL (Multi-Round Length) feature. You can extract the unnormalized embedding from llama.cpp's output and manually adjust (truncate or project) the embedding dimensions as needed.
> 
> 必须在embeddings = F.normalize(embeddings, p=2, dim=1)之前截断是吧

是的

**Comment by datalee** [comment]
像这样是吧：
`
embeddings = embeddings[:, :128]
`

**Comment by wkzcml-1** [comment]
> Sorry, llama.cpp does not currently support the MRL (Multi-Round Length) feature. You can extract the unnormalized embedding from llama.cpp's output and manually adjust (truncate or project) the embedding dimensions as needed.

So, isn't MRL the abbreviation for Matryoshka Representation Learning? Could you add some examples to illustrate how to use this model MRL?

**Comment by wkzcml-1** [comment]
> Sorry, llama.cpp does not currently support the MRL (Multi-Round Length) feature. You can extract the unnormalized embedding from llama.cpp's output and manually adjust (truncate or project) the embedding dimensions as needed.

So, isn't MRL the abbreviation for **Matryoshka Representation Learning**? Could you add some examples to illustrate how to use this model MRL?

**Comment by afalf** [comment]
> > Sorry, llama.cpp does not currently support the MRL (Multi-Round Length) feature. You can extract the unnormalized embedding from llama.cpp's output and manually adjust (truncate or project) the embedding dimensions as needed.
> 
> So, isn't MRL the abbreviation for **Matryoshka Representation Learning**? Could you add some examples to illustrate how to use this model MRL?

You can see ours examples [example](https://github.com/QwenLM/Qwen3-Embedding/blob/main/examples/qwen3_embedding_transformers.py).
